var/spack/repos/builtin/packages/saga-gis/package.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+
+
+class SagaGis(AutotoolsPackage):
+    """
+    SAGA is a GIS for Automated Geoscientific Analyses and has been designed
+    for an easy and effective implementation of spatial algorithms. It offers
+    a comprehensive, growing set of geoscientific methods and provides an
+    easily approachable user interface with many visualisation options
+    """
+    homepage    = "http://saga-gis.org/"
+    url         = "https://sourceforge.net/projects/saga-gis/files/SAGA%20-%205/SAGA%20-%205.0.0/saga-5.0.0.tar.gz"
+    git         = "git://git.code.sf.net/p/saga-gis/code"
+
+    version('develop',  branch='master')
+    version('7.0.0',    branch='release-7.0.0', preferred=True)
+    version('6.4.0',    branch='release-6.4.0')
+    version('6.3.0',    branch='release-6.3.0')
+    version('6.2.0',    branch='release-6.2.0')
+    version('6.1.0',    branch='release-6.1.0')
+    version('6.0.0',    branch='release-6.0.0')
+    version('5.0.1',    branch='release-5-0-1')
+    version('5.0.0',    branch='release-5.0.0')
+    version('4.1.0',    branch='release-4.1.0')
+    version('4.0.0',    branch='release-4.0.0')
+    version('4.0.0',    branch='release-4.0.0')
+    version('3.0.0',    branch='release-3.0.0')
+    version('2.3-lts',  branch='release-2-3-lts')
+    version('2.3.1',    branch='release-2-3-1')
+    version('2.3.0',    branch='release-2-3-0')
+
+    variant('gui',      default=True,   description='Build GUI and interactive SAGA tools')
+    variant('odbc',     default=True,   description='Build with ODBC support')
+    variant('triangle', default=True,   description='Build with triangle.c non free for commercial use otherwise use qhull')
+    variant('libfire',  default=True,   description='Build with libfire (non free for commercial usage)')
+    variant('openmp',   default=True,   description='Build with OpenMP enabled')
+    variant('python',   default=False,  description='Build Python extension')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    # FIXME unnecessary dependency on python3 because of implicit python3
+    # dependency through meson by a dependency of wx/gtkplus
+    depends_on('python@3:')
```

Comment:
Is there any reason you have to specify `python@3:`? Doesn't wx/gtkplus do that for you?

Reply:
Without this dependency the concretizer fails, since another dependency requires by default python2:

    $ spack spec saga-gis
    Input spec
    --------------------------------
    saga-gis
    
    Concretized
    --------------------------------
    ==> Error: An unsatisfiable version constraint has been detected for spec:
    
        python@2.7.15%gcc@8.2.1+dbm~optimizations+pic+pythoncmd+shared~tk~ucs4 arch=linux-fedora28-x86_64 
            ^bzip2@1.0.6%gcc@8.2.1+shared arch=linux-fedora28-x86_64 
                ^diffutils@3.6%gcc@8.2.1 arch=linux-fedora28-x86_64 
            ^gdbm@1.18.1%gcc@8.2.1 arch=linux-fedora28-x86_64 
                ^readline@7.0%gcc@8.2.1 arch=linux-fedora28-x86_64 
                    ^ncurses@6.1%gcc@8.2.1~symlinks~termlib arch=linux-fedora28-x86_64 
                        ^pkgconf@1.5.4%gcc@8.2.1 arch=linux-fedora28-x86_64 
            ^openssl@1.1.1%gcc@8.2.1+systemcerts arch=linux-fedora28-x86_64 
                ^perl@5.26.2%gcc@8.2.1+cpanm+shared+threads arch=linux-fedora28-x86_64 
                ^zlib@1.2.11%gcc@8.2.1+optimize+pic+shared arch=linux-fedora28-x86_64 
            ^sqlite@3.25.3%gcc@8.2.1~functions arch=linux-fedora28-x86_64 
    
    
    while trying to concretize the partial spec:
    
        meson
    
    
    meson requires python version 3:, but spec asked for 2.7.15

So one would need to install saga with `spack install saga-gis^python@3:` instead